src/components/Header/index.js

```diff
@@ -10,10 +11,75 @@ import {ReactComponent as MoreIcon} from 'assets/images/icons/more.svg'
 import {ReactComponent as LogInIcon} from 'assets/images/icons/logIn.svg'
 import {ReactComponent as RegisterIcon} from 'assets/images/icons/register.svg'
 import HeaderDropdown from 'components/Dropdown/HeaderDropdown'
+import * as productActions from 'redux/actions/productActions'
 
 import './styles.scss'
 
 const Header = () => {
+  const dispatch = useDispatch()
+  const navigate = useNavigate()
+
+  const [value, setValue] = useState('')
+  const [showDropDown, setShowDropDown] = useState(false)
+
+  const productFilterByName = useSelector(
+    state => state.Product.productFilterByName.data,
+  )
+
+  const onChangeInputSearch = e => {
+    setValue(e.target.value)
+    dispatch(productActions.loadProductFilterByName(e.target.value))
+  }
+
+  const onEnterSearch = e => {
+    if (e.keyCode === 13) {
+      e.preventDefault()
+      navigate(`/name/${e.target.value}/products`)
+      setValue('')
+      setShowDropDown(true)
+    }
+  }
+
+  const onFocusOutSearch = () => {
+    setShowDropDown(true)
+  }
+
+  const onFocusInSearch = () => {
+    setShowDropDown(false)
+  }
+
+  const onSetInputValue = id => {
+    setValue('')
+    navigate(`/product/${id}`)
+  }
+
+  const renderSearchDropdown = () => {
+    if (productFilterByName !== undefined) {
```

Comment:
this logic should be placed outside this scope, st like this on render:
{productFilterByName !== undefined && renderSearchDropdown()}
This avoid return an confused empty string